### PR TITLE
Check file types by header instead of by extension.

### PR DESCRIPTION
**Pull request by q82419**

### CODE REVIEW

File: lib/loader/ldmgr.cpp [hunk 1]
```diff
@@ -14,12 +14,9 @@ Expect<void> LDMgr::setPath(const std::filesystem::path &FilePath) {
   }
 
   const auto IntrinsicsTable = getSymbol<const void *>("intrinsics");
-  if (unlikely(!IntrinsicsTable)) {
```
Comment (hydai) [inline lib/loader/ldmgr.cpp:17]: I think we should keep this error checker.

**Review by hydai — CHANGES_REQUESTED** [review]

File: lib/loader/ldmgr.cpp (hunk 1 above)
Comment (q82419) [reply lib/loader/ldmgr.cpp:17]: This is for the case of AOT-WASM without intrinsics table.

**Comment by q82419** [comment]
@ibmibmibm  Please help to double check that the AOT core tests for universal WASM run in AOT mode, not interpreter mode.

File: tools/wasmedge/wasmedgec.cpp [hunk 2]
```diff
@@ -155,9 +155,9 @@ int main(int Argc, const char *Argv[]) {
     if (ConfGenericBinary.value()) {
       Conf.getCompilerConfigure().setGenericBinary(true);
     }
-    if (OutputPath.extension().u8string() == ".wasm"sv) {
+    if (OutputPath.extension().u8string() == ".so"sv) {
```
Comment (hydai) [inline tools/wasmedge/wasmedgec.cpp:158]: Looks like this is a breaking change in wasmedgec. We should add it on changelog.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]

**Pull request merged**